### PR TITLE
EWPP-3479: Add optional expandable block to Text with featured media pattern.

### PR DESCRIPTION
**Pull request by 22Alexandra**

### CODE REVIEW

**Review by sergepavle — DISMISSED** [review]

**Review by nagyad — CHANGES_REQUESTED** [review]
There are errors on the event form and event page. Not sure if it is related. 
image

File: tests/src/PatternAssertions/TextFeaturedMediaAssert.php [hunk 1]
```diff
@@ -121,6 +124,35 @@ protected function assertHighlighted(bool $highlighted, Crawler $crawler) {
...
+    if (!$expected_block) {
+      $this->assertElementNotExists('div.ecl-expandable.ecl-media-container__expandable', $crawler);
+      return;
+    }
+    // Assert the expandable selector.
+    $this->assertElementExists('div.ecl-expandable.ecl-media-container__expandable', $crawler);
+    // Assert the button with its toggle labels and icon.
+    $this->assertElementExists('button.ecl-button.ecl-button--secondary.ecl-expandable__toggle', $crawler);
+    $svg = $crawler->filter('span.ecl-button__container svg.ecl-icon.ecl-icon--fluid.ecl-icon--rotate-180.ecl-button__icon.ecl-button__icon--after use');
+    self::assertStringContainsString('icons.svg#corner-arrow', $svg->attr('xlink:href'));
```
Comment (nagyad) [inline tests/src/PatternAssertions/TextFeaturedMediaAssert.php]: Why not $this-> ? You are not in a static context.

File: tests/src/PatternAssertions/TextFeaturedMediaAssert.php [hunk 2]
```diff
@@ -121,6 +124,35 @@ protected function assertHighlighted(bool $highlighted, Crawler $crawler) {
...
+    $this->assertElementExists('button[data-ecl-label-expanded="' . $expected_block['label_expanded'] . '"][data-ecl-label-collapsed="' . $expected_block['label_collapsed'] . '"]', $crawler);
+    // Assert the content id and its text.
+    if (isset($expected_block['id'])) {
+      $this->assertElementExists('div#' . $expected_block['id'] . '-content.ecl-expandable__content', $crawler);
+    }
+    $content = $crawler->filter('div.ecl-expandable__content');
+    self::assertEquals($expected_block['content'], $content->text());
```
Comment (nagyad) [inline tests/src/PatternAssertions/TextFeaturedMediaAssert.php]: Why not $this-> ? You are not in a static context.

**Review by nagyad — CHANGES_REQUESTED** [review]

**Comment by 22Alexandra** [comment]
> There are errors on the event form and event page. Not sure if it is related. image

They're not related and they are generated by the helptext added by the Media modules regarding the allowed media types:
```Type part of the media name. See the [media list](http://127.0.0.1:8080/build/en/admin/content/media) (opens a new window) to help locate media. Allowed media types: AV Portal Photo, AV Portal Video, Image, Remote video, ,``` 
Basically the Iframe and Video iframe media types are allowed in the field but the module creating them is not installed. I fixed the dependencies in oe_content.

**Review by nagyad — DISMISSED** [review]

**Review by sergepavle — DISMISSED** [review]

**Review by sergepavle — DISMISSED** [review]

**Review by nagyad — APPROVED** [review]

**Review by sergepavle — APPROVED** [review]

**Pull request merged**